pkg/provisioner/common.go

```diff
@@ -0,0 +1,248 @@
+package provisioner
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	ghwutil "github.com/jaypipes/ghw/pkg/util"
+	"github.com/sirupsen/logrus"
+
+	diskv1 "github.com/harvester/node-disk-manager/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/node-disk-manager/pkg/block"
+)
+
+type NeedMountUpdateOP int8
+
+const (
+	TypeLonghornV1 = "LonghornV1"
+	TypeLonghornV2 = "LonghornV2"
+	TypeLVM        = "LVM"
+
+	// longhorn disk tags
+	ErrorCacheDiskTagsNotInitialized = "CacheDiskTags is not initialized"
+
+	// longhorn MountStatus
+	NeedMountUpdateNoOp NeedMountUpdateOP = 1 << iota
+	NeedMountUpdateMount
+	NeedMountUpdateUnmount
+)
+
+func (f NeedMountUpdateOP) Has(flag NeedMountUpdateOP) bool {
+	return f&flag != 0
+}
+
+type Provisioner interface {
+	// Format is the Prerequisites for the provisioner
+	// You should do the format-related operations (including mkfs, mount ...etc) here
+	// Return values: bool1: isFormatComplete, bool2: isNeedRequeue, error: error
```

Comment:
Does this sound better? If changed, adapt the following occurrences as well.
```suggestion
	// Return values: bool1: isFormatComplete, bool2: isRequeueNeeded, error: error
```

Reply:
done